Add support for HABv4 Fast Authentication

tientuk3 opened the pull request:
### Summary
Added support for generating HABv4 Fast Authentication compatible images using the "Install NOCAK" command in the BD configuration file.

### Background
The fast authentication mode (Install NOCAK command) for HAB containers was not available, although according to the documentation, nxpimage should fully support the legacy Elftosb configurations.

### Solution
Added support for the aforementioned command in components related to nxpimage, using the legacy Code Signing Tool source code, Code-Signing Tool User's Guide, and i.MXRT1060 Manufacturing User's Guide as references.

### Testing
Verified correct HAB secure boot operation on an i.MXRT1061 device, using this feature. 
Added a few test cases to the existing nxpimage HAB tests.
Verified using _codecheck.py_ that my changes did not generate any new test warnings.

### Additional Notes
Maybe it would be useful for the documentation to reflect this functionality in the future, since it is very useful in HSM based setups.

BohdanMarek commented on spsdk/image/hab/commands/commands.py:
```diff
@@ -291,7 +298,7 @@ def load_from_config(cls, config: HabConfig, search_paths: Optional[List[str]] =
         except SPSDKValueError as exc:
             # Keep the backwards compatibility with CSF tool and try to determine the path from certificate path
-            private_key_path = determine_private_key_path(install_csfk_params["InstallCSFK_File"])
+            private_key_path = determine_private_key_path(install_csfk_params[cfsk_path_param])
```
```suggestion
            private_key_path = determine_private_key_path(install_csfk_params[cert_path_param])
```

BohdanMarek commented on spsdk/image/hab/commands/commands.py:
```diff
@@ -380,9 +387,16 @@ def load_from_config(cls, config: HabConfig, search_paths: Optional[List[str]] =
+        # determine the key path, depending on if HAB is configured in normal or fast authentication mode
+        try:
+            install_key_params = config.commands.get_command_params(SecCommand.INSTALL_KEY)
+            cert_path_param = "InstallKey_File"
```
```suggestion
            key_path_param = "InstallKey_File"
```

BohdanMarek commented on spsdk/image/hab/commands/commands.py:
```diff
@@ -380,9 +387,16 @@ def load_from_config(cls, config: HabConfig, search_paths: Optional[List[str]] =
+            cert_path_param = "InstallKey_File"
+        except SPSDKValueError:  # in case of Fast Authentication, look for the NOCAK command
+            install_key_params = config.commands.get_command_params(SecCommand.INSTALL_NOCAK)
+            cert_path_param = "InstallNOCAK_File"
```
```suggestion
            key_path_param = "InstallNOCAK_File"
```

BohdanMarek commented:
Hi @tientuk3 ,
Thank you for your contribution. Unfortunately I found that the github pipeline does not run. Please until I fix it, run the codecheck.py script manually.

tientuk3 commented:
> Hi @tientuk3 , Thank you for your contribution. Unfortunately I found that the github pipeline does not run. Please until I fix it, run the codecheck.py script manually.

Thanks for the review comments, I addressed the issues in the latest commit (named the variables `cert_path_param` in both cases since they point to the public key certificates, this was a mistake I made earlier). I ran codecheck.py again manually, producing the same number of warnings as the current master:

![image](https://github.com/nxp-mcuxpresso/spsdk/assets/73032423/ed034195-2185-4c89-8659-91ec131feaf5)

BohdanMarek commented:
Hi @tientuk3,
As the github pipelines do not work right now, I merged your changes into our internal repo. The most recent release 2.1.1 already contains it. Can you take a look at it please? If everything is fine, we can close this PR.

tientuk3 commented:
> Hi @tientuk3, As the github pipelines do not work right now, I merged your changes into our internal repo. The most recent release 2.1.1 already contains it. Can you take a look at it please? If everything is fine, we can close this PR.

Hi, I tested the feature using release 2.1.1 and it seems to work fine. For me it's OK to close this PR. Thanks for letting me contribute 👍

BohdanMarek commented:
Thank you for your contribution. We will try to fix the github actions soon, so the pull requests can be merged directly to GH repo.
